static/selector.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable quotes */
+/**
+ * Creates the selectors for the chart and updates the chart based on the
+ * selected values.
+ * @param {string} url The url for the application.
+ * @param {string} chartId The id of the chart being drawn.
+ * @param {*} colorScale A d3 colorscale used for the line colors.
+ */
+const selectors = async (url, chartId, colorScale) => {
+  const modes = ["Default", "K-means", "DBSCAN"];
+  const similarity = ["Correlation", "Proximity"];
+  let clusters = ["All"];
+  createSelector("mode", modes, updateChart);
+  createSelector("similarity", similarity, updateChart);
+  createSelector("cluster", clusters, updateCluster);
+
+  /**
+   * Updates the chart according to the values of the selectors.
+   */
+  async function updateChart() {
+    const currentMode = d3.select("select#modeSelector").property("value");
+    const currentSimilarity = d3.select("select#similaritySelector")
+        .property("value");
+    createSelector("cluster", ["All"], updateCluster);
```

Comment:
Does it make sense to recreate the whole selector here? Seems like you're using this to just set the value. Can we create a function to set the value of the selector?

Reply:
The name createSelector might be confusing. The selector is created in index.html, createSelector is updating the options for the selector with a join.